src/main.rs

```diff
@@ -14,38 +15,34 @@ mod git;
 mod cargo;
 
 fn execute(args: &ArgMatches) -> Result<i32, error::FatalError> {
-    let cargo_file = try!(config::parse_cargo_config());
+    let properties: Table = try!(config::parse_config());
 
-    // Verify the TOML configuration if present.
-    if let Some(invalid_keys) = config::verify_release_config(&cargo_file) {
-        for invalid_key in invalid_keys {
+    // Verify the TOML configuration.
+    let valid_keys = vec![config::SIGN_COMMIT,
+                          config::PUSH_REMOTE,
+                          config::DOC_BRANCH,
+                          config::DOC_COMMIT_MESSAGE];
```

Comment:
I tried that but vec was doing heap allocations somewhere. I know what you mean, though, I don't like this.

Good news is I woke up knowing what the problem is... we want there to be a struct in config.rs that wraps the Table and provides the constructor and accessors. Will be able to remove this validation check nicely into the constructor.r